File: auth-service/src/main/java/com/eirsteir/coffeewithme/authservice/domain/RoleTypeConverter.java
package com.eirsteir.coffeewithme.authservice.domain;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter(autoApply = true)
public class RoleTypeConverter implements AttributeConverter<RoleType, String> {

  @Override
  public String convertToDatabaseColumn(RoleType type) {
    return type == null ? null : type.name();
  }

  @Override
  public RoleType convertToEntityAttribute(String value) {
    if (value == null) return null;

    return RoleType.from(value);
  }
}

File: auth-service/src/main/kotlin/com/eirsteir/coffeewithme/authservice/domain/RoleTypeConverter.kt
package com.eirsteir.coffeewithme.authservice.domain

import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter(autoApply = true)
class RoleTypeConverter : AttributeConverter<RoleType, String> {

    override fun convertToDatabaseColumn(type: RoleType) = type.name

    override fun convertToEntityAttribute(value: String) = RoleType.from(value)

}